[BUG] License collector fails to execute the command

When License collector is enabled, results in error:
`collector.go:884: Error collecting license info: failed to execute xicli license show: failed to execute xicli command: exit status 3`

Collector tries to execute the command with unsupported option:
`xicli license show --format json`

xicli-esporter version: 1.1.0
xiraid version: 4.2
OS: Ubuntu 22.04

--- a/pkg/xicli/xicli.go
+++ b/pkg/xicli/xicli.go
@@ -9,6 +9,21 @@ import (
 	"time"
 )
 
+// FlexInt is an int that can be unmarshalled from either a JSON number or a
+// non-numeric string (e.g. "-"). Non-numeric strings are treated as zero.
+type FlexInt int
+
+func (f *FlexInt) UnmarshalJSON(data []byte) error {
+	var n int
+	if err := json.Unmarshal(data, &n); err == nil {
+		*f = FlexInt(n)
+		return nil
+	}
+	// Non-numeric value (e.g. "-"): treat as zero.
+	*f = 0
+	return nil
+}
+
 // Executor handles execution of xicli commands.
 type Executor struct {
 	xicliPath string
@@ -32,7 +47,7 @@ type RaidInfo struct {
 	DevicesHealth     []string        `json:"devices_health"`
 	DevicesWear       []string        `json:"devices_wear"`
 	GroupSize         int             `json:"group_size"`
-	MemoryUsageMB     int             `json:"memory_usage_mb"`
+	MemoryUsageMB     FlexInt         `json:"memory_usage_mb"`
 	MemoryPreallocMB  int             `json:"memory_prealloc_mb,omitempty"`
 	MemoryLimitMB     int             `json:"memory_limit_mb,omitempty"`
 	BlockSize         int             `json:"block_size"`

--- a/pkg/xicli/xicli.go
+++ b/pkg/xicli/xicli.go
@@ -15,12 +15,16 @@ type FlexInt int
 
 func (f *FlexInt) UnmarshalJSON(data []byte) error {
 	var n int
-	if err := json.Unmarshal(data, &n); err == nil {
+
+	err := json.Unmarshal(data, &n)
+	if err == nil {
 		*f = FlexInt(n)
+
 		return nil
 	}
 	// Non-numeric value (e.g. "-"): treat as zero.
 	*f = 0
+
 	return nil
 }
 

--- a/pkg/xicli/xicli.go
+++ b/pkg/xicli/xicli.go
@@ -53,7 +53,7 @@ type RaidInfo struct {
 	MergeWriteEnabled int             `json:"merge_write_enabled,omitempty"`
 }
 
-// LicenseInfo represents the output of 'xicli license show --format json'.
+// LicenseInfo represents the output of 'xicli license show'.
 type LicenseInfo struct {
 	Status        string `json:"status"`
 	Type          string `json:"type"`
@@ -178,9 +178,9 @@ func (x *Executor) GetFaultyCount() (map[string]int, error) {
 	return faultyCounts, nil
 }
 
-// GetLicenseInfo executes 'xicli license show --format json'.
+// GetLicenseInfo executes 'xicli license show'.
 func (x *Executor) GetLicenseInfo() (*LicenseInfo, error) {
-	output, err := x.executeCommand("license", "show", "--format", "json")
+	output, err := x.executeCommand("license", "show")
 	if err != nil {
 		return nil, fmt.Errorf("failed to execute xicli license show: %w", err)
 	}

